### PR TITLE
[SUBMARINE-185] Add how to contribution docs

### PR DESCRIPTION
**Pull request by liuxunorg**
### What is this PR for?
This document is intended to help more people and know-how to participate in the development of submarine projects.

The documentation includes:
1. Contribution Guide
2. Github Workflow
3. Code Review Guide

### What type of PR is it?
Improvement

### Todos
* [ ] - Need add become-a-committer.md
* [ ] - Need add contributors.md

### What is the Jira issue?
* https://issues.apache.org/jira/browse/SUBMARINE-185

### How should this be tested?
* [CI Pass](https://travis-ci.org/liuxunorg/hadoop-submarine/builds/588083937)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,13 +104,14 @@ This is a docker image built for submarine development and quick start test.
...
-
-
-
+Read the [Quick Start Guide](./docs/helper/QuickStart.md)
 
+## Apache Hadoop Submarine Community
 
+Read the [Apache Hadoop Submarine Community Guide](./docs/community/README.md)
 
+How to contributing [Contributing Guide](./docs/community/contributing.md)
```
Comment (jojochuang) [inline README.md]: How to contribute

File: README.md [hunk 2]
```diff
@@ -104,13 +104,14 @@ This is a docker image built for submarine development and quick start test.
...
 
+## License
 
+Apache Hadoop Submarine Community is under the Apache 2.0 license. See the [LICENSE](./LICENSE) file for details.
```
Comment (jojochuang) [inline README.md]: The Apache Hadoop Submarine project is licensed under the Apache 2.0 License.

File: docs/community/README.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
++ Slack User: [https://the-asf.slack.com/submarine-user/](https://the-asf.slack.com/submarine-user/)
+
++ Zoom: [https://cloudera.zoom.us/j/880548968](https://cloudera.zoom.us/j/880548968)
+
++ Sync Up: [https://docs.google.com/document/d/16pUO3TP4SxSeLduG817GhVAjtiph9HYpRHo_JgduDvw/edit](https://docs.google.com/document/d/16pUO3TP4SxSeLduG817GhVAjtiph9HYpRHo_JgduDvw/edit)
+
+## Your First Contribution
+
+You can start by finding an existing issue with the [https://issues.apache.org/jira/browse/SUBMARINE](https://issues.apache.org/jira/browse/SUBMARINE) label. These issues are well suited for new contributors.
```
Comment (jojochuang) [inline docs/community/README.md]: The links lists all SUBMARINE jiras, with no filters for labels.

File: docs/community/README.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+Once you become a Hadoop Submarine contributor, check your name here: [CONTRIBUTORS](contributors.md)
+
+Contributions are welcomed and greatly appreciated. See [CONTRIBUTING.md](contributing.md) for details on submitting patches and the contribution workflow.
+
+## How Do I Become a Committer?
+
+First of all you need to get involved and be a Contributor.
+
+Based on your track-record as a contributor, one of our Maintainers or PMC members may invite you to be a committer (after we've called a vote). When that happens, if you accept, the following process kicks into place...
```
Comment (jojochuang) [inline docs/community/README.md]: Per Apache code, PMCs vote on committership.

File: docs/community/contributing.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Contribution Guidelines
+
+**Apache Hadoop Submarine** is an [Apache2 License](https://github.com/apache/hadoop-submarine/blob/master/LICENSE) Software.
```
Comment (jojochuang) [inline docs/community/contributing.md]: Better spell as "Apache 2.0 License"

File: docs/community/contributing.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+
+Contributing to Hadoop Submarine (Source code, Documents, Image, Website) means you agree to the Apache2 License.
```
Comment (jojochuang) [inline docs/community/contributing.md]: Usually a contributor should submit a Contributor Agreement as well: https://www.apache.org/licenses/contributor-agreements.html

File: docs/community/contributing.md [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+```sh
+git commit
+```
+
+Likely you'll go back and edit/build/test further, and then `commit --amend` in a
+few cycles.
+
+### Step 7: Push
+
+When the changes are ready to review (or you just to create an offsite backup
```
Comment (jojochuang) [inline docs/community/contributing.md]: or you just "want" to

File: docs/community/contributing.md [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+or your work), push your branch to your fork on `github.com`:
```
Comment (jojochuang) [inline docs/community/contributing.md]: "of" your work

File: docs/community/contributing.md [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+
+    ### How should this be tested?
+    Outline the steps to test the PR here.
+
+    ### Screenshots (if appropriate)
+
+    ### Questions:
+    * Does the licenses files need update?
+    * Is there breaking changes for older versions?
+    * Does this needs documentation?
```
Comment (jojochuang) [inline docs/community/contributing.md]: Does this need documentation?

File: docs/community/contributing.md [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+    * Open an issue on Jira https://issues.apache.org/jira/browse/SUBMARINE/
+    * Put link here, and add [SUBMARINE-${jira_number}] in PR title, eg. [SUBMARINE-323]
+
+    ### How should this be tested?
+    Outline the steps to test the PR here.
+
+    ### Screenshots (if appropriate)
+
+    ### Questions:
+    * Does the licenses files need update?
```
Comment (jojochuang) [inline docs/community/contributing.md]: Do the licenses files need updates

File: docs/community/contributing.md [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+
+## The Review Process
+
+When a Pull Request is submitted, it is being merged or rejected by the  following review process.
+
+* Anybody can be a reviewer and may comment on the change or suggest modifications.
+* Reviewer can indicate that a patch looks suitable for merging with a comment such as: "Looks good", "LGTM", "+1".
+* At least one indication of suitability (e.g. "LGTM") from a committer is required to be merged.
+* Pull request is open for 1 or 2 days for potential additional review, unless it's got enough indication of suitability.
+* A committer can then initiate lazy consensus ("Merge if there is no more discussion") after what the code can be merged after a certain time (normally 24 hours) if there is no more reviews.
```
Comment (jojochuang) [inline docs/community/contributing.md]: after "what" --> which

File: README.md (hunk 2 above)
Comment (liuxunorg) [reply README.md]: done.

File: README.md (hunk 1 above)
Comment (liuxunorg) [reply README.md]: done.

File: docs/community/README.md (hunk 4 above)
Comment (liuxunorg) [reply docs/community/README.md]: done.

File: docs/community/contributing.md (hunk 5 above)
Comment (liuxunorg) [reply docs/community/contributing.md]: done.

File: docs/community/contributing.md (hunk 10 above)
Comment (liuxunorg) [reply docs/community/contributing.md]: done.

File: docs/community/contributing.md (hunk 9 above)
Comment (liuxunorg) [reply docs/community/contributing.md]: done.

File: docs/community/contributing.md (hunk 11 above)
Comment (liuxunorg) [reply docs/community/contributing.md]: done.

File: docs/community/contributing.md (hunk 7 above)
Comment (liuxunorg) [reply docs/community/contributing.md]: done.

File: docs/community/contributing.md (hunk 8 above)
Comment (liuxunorg) [reply docs/community/contributing.md]: done.

File: docs/community/README.md (hunk 3 above)
Comment (liuxunorg) [reply docs/community/README.md]: I don't know what URL to fill in, can you give me a URL address?

File: docs/community/contributing.md (hunk 6 above)
Comment (liuxunorg) [reply docs/community/contributing.md]: I have consulted this question, Apache 2.0 license, you can not sign Contributor Agreement.

**Comment by tangzhankun** [comment]
Thanks for @liuxunorg 's patch and @jojochuang 's review. I'll commit this if no more discussion.

**Pull request closed without merging**